src/shogun/neuralnets/NeuralNetwork.cpp

```diff
@@ -309,6 +309,18 @@ bool CNeuralNetwork::train_gradient_descent(SGMatrix<float64_t> inputs,
 
 			float64_t e = compute_gradients(inputs_batch, targets_batch, gradients);
 
+
+			for (int32_t k=0; k<m_num_layers; k++)
```

Comment:
@lisitsyn This for loop can be made a private member function if you want as it is getting repeated in lbfgs also.